app/src/common/gecko/com/igalia/wolvic/browser/api/impl/SessionImpl.java

```diff
@@ -0,0 +1,391 @@
+
+    @Override
+    public void reload(int flags) {
+        mSession.reload(toGeckoFlags(flags));
+    }
+
+    @Override
+    public void stop() {
+        mSession.stop();
+    }
+
+    @Override
+    public void setActive(boolean active) {
+        mSession.setActive(active);
+    }
+
+    @Override
+    public void setFocused(boolean focused) {
+        mSession.setFocused(focused);
+    }
+
+    @Override
+    public void open(@NonNull WRuntime runtime) {
+        mSession.open(((RuntimeImpl)runtime).getGeckoRuntime());
+    }
+
+    @Override
+    public boolean isOpen() {
+        return mSession.isOpen();
+    }
+
+    @Override
+    public void close() {
+        mSession.close();
+    }
+
+    @Override
+    public void goBack(boolean userInteraction) {
+        mSession.goBack();
+    }
+
+    @Override
+    public void goForward(boolean userInteraction) {
+        mSession.goForward();
+    }
+
+    @Override
+    public void gotoHistoryIndex(int index) {
+        mSession.gotoHistoryIndex(index);
+    }
+
+    @Override
+    public void purgeHistory() {
+        mSession.purgeHistory();
+    }
+
+    @NonNull
+    @Override
+    public WSessionSettings getSettings() {
+        return mSettings;
+    }
+
+    @Override
+    public void exitFullScreen() {
+        mSession.exitFullScreen();
+    }
+
+    @NonNull
+    @Override
+    public WDisplay acquireDisplay() {
+        return new DisplayImpl(mSession.acquireDisplay());
+    }
+
+    @Override
+    public void releaseDisplay(@NonNull WDisplay display) {
+        mSession.releaseDisplay(((DisplayImpl)display).getGeckoDisplay());
+    }
+
+    @Override
+    public void restoreState(@NonNull WSessionState state) {
+        mSession.restoreState(((SessionStateImpl)state).getGeckoState());
+    }
+
+    @Override
+    public void getClientToSurfaceMatrix(@NonNull Matrix matrix) {
+        mSession.getClientToSurfaceMatrix(matrix);
+    }
+
+    @Override
+    public void getClientToScreenMatrix(@NonNull Matrix matrix) {
+        mSession.getClientToScreenMatrix(matrix);
+    }
+
+    @Override
+    public void getPageToScreenMatrix(@NonNull Matrix matrix) {
+        mSession.getPageToScreenMatrix(matrix);
+    }
+
+    @Override
+    public void getPageToSurfaceMatrix(@NonNull Matrix matrix) {
+        mSession.getPageToSurfaceMatrix(matrix);
+    }
+
+    @NonNull
+    @Override
+    public WTextInput getTextInput() {
+        return mTextInput;
+    }
+
+    @NonNull
+    @Override
+    public WPanZoomController getPanZoomController() {
+        return mPanZoomController;
+    }
+
+    @Override
+    public void setContentDelegate(@Nullable ContentDelegate delegate) {
+        if (mContentDelegate == delegate) {
+            return;
+        }
+        mContentDelegate = delegate;
+        if (mContentDelegate == null) {
+            mSession.setContentDelegate(null);
+        } else {
+            mSession.setContentDelegate(new ContentDelegateImpl(mContentDelegate, this));
```

Comment:
What about using the ternary operator to avoid the if-else block and do it in one line?

Reply:
done